### PR TITLE
Add Snowbridge pallets

### PR DESCRIPTION
**Pull request by alistair-singh**
## Add Snowbridge pallets
Adds the beacon light client to the `bridge-hub-rococo-local` runtime.
## TODO
- [x] Add beacon client pallet.
- [x] Add relayer accounts
- [x] Re-generate weights for Beacon client.
- [x] Add inbound channel.
- [x] Add basic channel rpc

### CODE REVIEW

File: parachains/runtimes/bridge-hubs/bridge-hub-rococo/Cargo.toml [hunk 1]
```diff
@@ -126,6 +133,11 @@ std = [
 	"xcm-builder/std",
 	"xcm-executor/std",
 	"xcm/std",
+    "snowbridge-basic-channel/std",
```
Comment (vgeddes) [inline parachains/runtimes/bridge-hubs/bridge-hub-rococo/Cargo.toml]: tabs vs spaces

File: parachains/runtimes/bridge-hubs/bridge-hub-rococo/Cargo.toml [hunk 2]
```diff
@@ -144,6 +156,10 @@ runtime-benchmarks = [
 	"pallet-collator-selection/runtime-benchmarks",
 	"cumulus-pallet-xcmp-queue/runtime-benchmarks",
 	"pallet-xcm-benchmarks/runtime-benchmarks",
+	"ethereum-beacon-client/runtime-benchmarks",
+	"snowbridge-core/runtime-benchmarks",
+    "snowbridge-basic-channel/runtime-benchmarks",
```
Comment (vgeddes) [inline parachains/runtimes/bridge-hubs/bridge-hub-rococo/Cargo.toml]: ditto

File: polkadot-parachain/src/chain_spec/bridge_hubs.rs [hunk 3]
```diff
@@ -327,6 +327,10 @@ pub mod rococo {
 						get_account_id_from_seed::<sr25519::Public>("Dave//stash"),
 						get_account_id_from_seed::<sr25519::Public>("Eve//stash"),
 						get_account_id_from_seed::<sr25519::Public>("Ferdie//stash"),
+						// Ethereum Bridge Relayer Accounts
+						get_account_id_from_seed::<sr25519::Public>("Relay"),
```
Comment (vgeddes) [inline polkadot-parachain/src/chain_spec/bridge_hubs.rs]: Should only need 2 of these 3 accounts. I think one of them's obsolete. Can you check?

**Review by vgeddes — COMMENTED** [review]
Looks good!

Besides my inline comments, I suggest the following:

In our cumulus fork, create an integration branch named `snowbridge`, and update Github to set it as the default branch. Then update this PR to merge into that.

Then in a month's time when we are ready to merge into upstream, we merge `snowbridge` into upstream master`.

**Comment by alistair-singh** [comment]
> In our cumulus fork, create an integration branch named `snowbridge` and update Github to set it as the default branch. Then update this PR to merge into that.

I have added the `snowbridge` branch was branched from `release-v0.9.380`.

File: polkadot-parachain/src/chain_spec/bridge_hubs.rs (hunk 3 above)
Comment (alistair-singh) [reply polkadot-parachain/src/chain_spec/bridge_hubs.rs]: `Relay` is not needed. There is an old script that references it. Specifically the `SubBeefy` command in `snowbridge/relayer/magefile.go`.
Removed in https://github.com/Snowfork/snowbridge/pull/785#:~:text=across%20the%20bridge.-,removed%20sub%20beefy%20magefile%20command,-8104ecc

File: scripts/benchmarks-ci.sh [hunk 4]
```diff
@@ -64,6 +64,10 @@ elif [[ $runtimeName == "bridge-hub-rococo" ]]; then
                 cumulus_pallet_xcmp_queue
                 pallet_xcm_benchmarks::generic
                 pallet_xcm_benchmarks::fungible
+                # snowbridge_basic_channel::inbound
+                snowbridge_basic_channel::outbound
+                # snowbridge_dispatch
```
Comment (alistair-singh) [inline scripts/benchmarks-ci.sh:69]: `snowbridge_basic_channel::inbound` and `snowbridge_dispatch` require benchmarking.

**Review by vgeddes — APPROVED** [review]
Looks good 🚀

**Pull request merged**